nixos/modules/services/hardware/asusctl.nix

```diff
@@ -0,0 +1,482 @@
+          standard = ["Static", "Breathe", "Strobe", "Rainbow", "Pulse"]
+          multizone = []
+          per_key = false
+
+          [[led_data]]
+          prod_family = "ROG Flow X16"
+          board_names = ["GV601RW", "GV601RM"]
+          standard = ["Static", "Breathe", "Strobe", "Pulse"]
+          multizone = []
+          per_key = false
+        '';
+      };
+
+      package = mkPackageOption pkgs "asusctl" {
+        default = [ "asusctl" ];
+      };
+
+      autoStartRogControlCenter = mkOption {
+        default = true;
+        type = types.bool;
+        description = lib.mdDoc "Whether rog-control-center should be started automatically.";
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [
+      cfg.package
+      (mkIf cfg.autoStartRogControlCenter (pkgs.makeAutostartItem { name = "rog-control-center"; package = cfg.package; }))
+    ];
+
+    systemd.services.asusd = {
+      description = "ASUS Notebook Control";
+      wantedBy = [ "multi-user.target" ];
+      before = [ "multi-user.target" ];
+      startLimitIntervalSec = 200;
+      startLimitBurst = 2;
+
+      serviceConfig = {
+        Environment = "IS_SERVICE=1";
+        ExecStartPre = "${pkgs.coreutils}/bin/sleep 2";
+        ExecStart = "${cfg.package}/bin/asusd";
+        Restart = "on-failure";
+        RestartSec = 1;
+        Type = "dbus";
+        BusName = "org.asuslinux.Daemon";
+        SELinuxContext = "system_u:system_r:unconfined_t:s0";
+      };
+    };
+
+    systemd.user.services.asus-notify = {
+      description = "ASUS Notifications";
+      wantedBy = [ "default.target" ];
+      startLimitIntervalSec = 200;
+      startLimitBurst = 2;
+
+      serviceConfig = {
+        ExecStartPre = "${pkgs.coreutils}/bin/sleep 2";
+        ExecStart = "${cfg.package}/bin/asus-notify";
+        Restart = "on-failure";
+        RestartSec = 1;
+        Type = "simple";
+      };
+    };
+
+    systemd.user.services.asusd-user = mkIf cfg.enableUserService {
+      description = "ASUS User Daemon";
+      wantedBy = [ "default.target" ];
+      startLimitIntervalSec = 200;
+      startLimitBurst = 2;
+
+      serviceConfig = {
+        ExecStartPre = "${pkgs.coreutils}/bin/sleep 2";
+        ExecStart = "${cfg.package}/bin/asusd-user";
+        Restart = "on-failure";
+        RestartSec = 1;
+        Type = "simple";
+      };
+    };
+
+    environment.etc."asusd/anime.conf" = {
+      source = animeConfigFile;
+      mode = "0644";
+    };
+    environment.etc."asusd/asusd.conf" = {
+      source = asusdConfigFile;
+      mode = "0644";
+    };
+    environment.etc."asusd/aura.conf" = {
+      source = auraConfigFile;
+      mode = "0644";
+    };
+    environment.etc."asusd/profile.conf" = {
+      source = profileConfigFile;
+      mode = "0644";
+    };
+    environment.etc."asusd/asusd-ledmodes.toml" = {
+      source = ledModesConfigFile;
+      mode = "0644";
```

Comment:
Can't we leave the default mode? The location is not writtable anyway

Reply:
Unfortunately no. The daemon writes those files on request by the rog-control-center binary provided in the package modifies those files when it is started.

Reply:
I am not sure if you can write to those files even with 644. Also nix overwrites them when switching/restarting.

Reply:
You definitely can't.